ui/snippets/walletMenu/WalletMenuDesktop.tsx

```diff
@@ -0,0 +1,110 @@
+import type { ButtonProps } from '@chakra-ui/react';
+import { Popover, PopoverContent, PopoverBody, PopoverTrigger, Button, useColorModeValue, Box, useBoolean, Tooltip } from '@chakra-ui/react';
+import React from 'react';
+
+import AddressIdenticon from 'ui/shared/entities/address/AddressIdenticon';
+import HashStringShorten from 'ui/shared/HashStringShorten';
+import useWallet from 'ui/snippets/walletMenu/useWallet';
+import WalletMenuContent from 'ui/snippets/walletMenu/WalletMenuContent';
+
+type Props = {
+  isHomePage?: boolean;
+};
+
+const WalletMenuDesktop = ({ isHomePage }: Props) => {
+  const { isWalletConnected, address, connect, disconnect, isModalOpening, isModalOpen } = useWallet();
+  const [ isPopoverOpen, setIsPopoverOpen ] = useBoolean(false);
+  const [ isTooltipShown, setIsTooltipShown ] = useBoolean(false);
+
+  React.useEffect(() => {
+    const wasShown = window.localStorage.getItem('wallet-connect-tooltip-shown');
+    if (!wasShown) {
+      setIsTooltipShown.on();
+      window.localStorage.setItem('wallet-connect-tooltip-shown', 'true');
+    }
+  }, [ setIsTooltipShown ]);
+
+  const variant = React.useMemo(() => {
+    if (isWalletConnected) {
+      return 'subtle';
+    }
+    return isHomePage ? 'solid' : 'outline';
+  }, [ isWalletConnected, isHomePage ]);
+
+  let buttonStyles: Partial<ButtonProps> = {};
+  const themedBackground = useColorModeValue('blackAlpha.50', 'whiteAlpha.50');
+  const themedBorderColor = useColorModeValue('gray.300', 'gray.700');
+  const themedColor = useColorModeValue('blackAlpha.800', 'gray.400');
```

Comment:
I see that these styles are repeated in 4 different components. Can we move them, for example, to a utility hook?